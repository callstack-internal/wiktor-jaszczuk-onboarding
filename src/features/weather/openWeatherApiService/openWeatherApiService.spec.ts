import {OpenWeatherApiService} from './index';

describe('OpenWeatherApiService', () => {
  it('the class has instance getter which returns singleton', () => {
    const instance1 = OpenWeatherApiService.getInstance();
    const instance2 = OpenWeatherApiService.getInstance();

    expect(instance2).toBe(instance1);
  });

  it('resolveCoordinatesByLocationName returns some data', async () => {
    await expect(
      OpenWeatherApiService.getInstance().resolveCoordinatesByLocationName(
        'WrocÅ‚aw',
        'pl',
      ),
    ).resolves.toMatchSnapshot();
  });

  it('getWeatherForLocation returns some data', async () => {
    await expect(
      OpenWeatherApiService.getInstance().getWeatherForLocation(17.76, -65.682),
    ).resolves.toMatchSnapshot();
  });
});
