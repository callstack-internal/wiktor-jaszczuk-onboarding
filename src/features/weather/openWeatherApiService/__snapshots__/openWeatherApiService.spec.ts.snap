// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OpenWeatherApiService getWeatherForLocation returns some data 1`] = `
{
  "clouds": {
    "all": 0,
  },
  "coord": {
    "lat": 51.11,
    "lon": 17.03,
  },
  "dt": 1741185257,
  "main": {
    "feels_like": 15.48,
    "grnd_level": 1008,
    "humidity": 23,
    "pressure": 1024,
    "sea_level": 1024,
    "temp": 17.11,
    "temp_max": 17.21,
    "temp_min": 16.69,
  },
  "sys": {
    "country": "PL",
    "sunrise": 1741152487,
    "sunset": 1741192736,
  },
  "timezone": 3600,
  "visibility": 10000,
  "weather": [
    {
      "description": "clear sky",
      "icon": "01d",
      "id": 800,
      "main": "Clear",
    },
  ],
  "wind": {
    "deg": 270,
    "speed": 2.57,
  },
}
`;

exports[`OpenWeatherApiService resolveCoordinatesByLocationName returns some data 1`] = `
{
  "country": "PL",
  "lat": 51.1089776,
  "local_names": {
    "ar": "فروتزواف",
    "be": "Уроцлаў",
    "bg": "Вроцлав",
    "cs": "Vratislav",
    "de": "Breslau",
    "en": "Wrocław",
    "eo": "Vroclavo",
    "es": "Breslavia",
    "fr": "Vratislavie",
    "he": "ורוצלב",
    "hu": "Boroszló",
    "it": "Breslavia",
    "ja": "ヴロツワフ",
    "la": "Wratislavia",
    "lt": "Vroclavas",
    "lv": "Vroclava",
    "mk": "Вроцлав",
    "pl": "Wrocław",
    "pt": "Breslávia",
    "ru": "Вроцлав",
    "sk": "Vroclav",
    "sr": "Вроцлав",
    "uk": "Вроцлав",
    "yi": "ברעסלוי",
    "zh": "弗罗茨瓦夫",
  },
  "lon": 17.0326689,
  "name": "Wrocław",
  "state": "Lower Silesian Voivodeship",
}
`;
